scripts/ROTD-JP/c101101013.lua

```diff
@@ -0,0 +1,72 @@
+--焰圣骑士-奥吉尔
+function c101101013.initial_effect(c)
+    --tograve
+    local e1=Effect.CreateEffect(c)
+    e1:SetDescription(aux.Stringid(101101013,0))
+    e1:SetCategory(CATEGORY_TOGRAVE)
+    e1:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+    e1:SetCode(EVENT_SUMMON_SUCCESS)
+    e1:SetCountLimit(1,101101013)
+    e1:SetTarget(c101101013.tgtg)
+    e1:SetOperation(c101101013.tgop)
+    c:RegisterEffect(e1)
+    local e2=e1:Clone()
+    e2:SetCode(EVENT_SPSUMMON_SUCCESS)
+    c:RegisterEffect(e2)
+    --equip
+    local e3=Effect.CreateEffect(c)
+    e3:SetCategory(CATEGORY_EQUIP)
+    e3:SetType(EFFECT_TYPE_IGNITION)
+    e3:SetProperty(EFFECT_FLAG_CARD_TARGET)
+    e3:SetRange(LOCATION_GRAVE)
+    e3:SetCountLimit(1,101101113)
+    e3:SetTarget(c101101013.eqtg)
+    e3:SetOperation(c101101013.eqop)
+    c:RegisterEffect(e3)
+    --indes
+    local e4=Effect.CreateEffect(c)
+    e4:SetType(EFFECT_TYPE_EQUIP)
+    e4:SetCode(EFFECT_INDESTRUCTABLE_EFFECT)
+    e4:SetValue(1)
+    c:RegisterEffect(e4)
+end
+function c101101013.tgfilter(c)
+    return ((c:IsRace(RACE_WARRIOR) and c:IsAttribute(ATTRIBUTE_FIRE)) or c:IsSetCard(0x207a)) and not c:IsCode(101101013) and c:IsAbleToGrave()
+end
+function c101101013.tgtg(e,tp,eg,ep,ev,re,r,rp,chk)
+    if chk==0 then return Duel.IsExistingMatchingCard(c101101013.tgfilter,tp,LOCATION_DECK,0,1,nil) end
+    Duel.SetOperationInfo(0,CATEGORY_TOGRAVE,nil,1,tp,LOCATION_DECK)
+end
+function c101101013.tgop(e,tp,eg,ep,ev,re,r,rp)
+    Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_TOGRAVE)
+    local g=Duel.SelectMatchingCard(tp,c101101013.tgfilter,tp,LOCATION_DECK,0,1,1,nil)
+    if g:GetCount()>0 then
+        Duel.SendtoGrave(g,REASON_EFFECT)
+    end
+end
+function c101101013.eqtg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+    if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) and chkc:IsFaceup() end
+    if chk==0 then return Duel.GetLocationCount(tp,LOCATION_SZONE)>0
+        and Duel.IsExistingTarget(Card.IsRace,tp,LOCATION_MZONE,0,1,nil,RACE_WARRIOR) end
+    Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_EQUIP)
+    Duel.SelectTarget(tp,Card.IsRace,tp,LOCATION_MZONE,0,1,1,nil,RACE_WARRIOR)

```

Comment:
suggested change:
1.
`if chk==0 then return Duel.GetLocationCount(tp,LOCATION_SZONE)>0
        and Duel.IsExistingTarget(Card.IsRace,tp,LOCATION_MZONE,0,1,nil,RACE_WARRIOR)
        and e:GetHandler():IsRelateToEffect(e) end`
2.
`Duel.SetOperationInfo(0,CATEGORY_LEAVE_GRAVE,e:GetHandler(),1,0,0)`
I don't know if it's necessary to change the first one because it may come true in core.